### DISCUSSION TITLE
Local installation?!

### DISCUSSION THREAD

**Discussion by jhmonroe** [givanz/Vvveb#250] [General]
Can anyone who has set this up locally help me get started with playing around with this? The installation instructions on the official site are very vague and I am getting blank error pages when trying to set up using MAMP. the installer does not just come up automatically. i messed with the conf file, tried copying it into the apache folder, etc

**Comment by givanz** [comment]
For  MAMP and XAMPP you don't need to change anything for apache, you only need to unzip `latest.zip` to htdocs folder.

Can you please check `C:\MAMP\logs\php_error.log` to see if there are any errors?

![mamp-5](https://github.com/user-attachments/assets/e381cd68-7fa1-4e29-a743-4297c7fd81ad)


I installed using MAMP and didn't encouter errors during install.
I downloaded latest.zip from https://www.vvveb.com and unziped inside `C:\MAMP\htdocs`
Then I accessed http://localhost choose sqlite and clicked install 


![mamp-1](https://github.com/user-attachments/assets/e4c1e18a-ee3c-4f0a-8a14-6baeddefc880)
![mamp-2](https://github.com/user-attachments/assets/90830580-9010-4578-ad3b-0d46372d19cf)
![mamp-3](https://github.com/user-attachments/assets/08e12f31-2c40-46ea-8146-d231685438a3)
![mamp-4](https://github.com/user-attachments/assets/2947bf9b-242e-4a30-8088-3f4ca15f7e8f)

**Comment by jhmonroe** [comment]
Hey! Thanks for replying. 
I tried with WP Local and MAMP. I tried in a subfolder (named vvveb) and also the root.
My MAMP uses a URL of http://localhost:8888 (with the 8888) which then redirects to http://localhost:8888/install/index.php and I receive the following error on the Browser side:

Internal Server Error
The server encountered an internal error or misconfiguration and was unable to complete your request.
Please contact the server administrator at you@example.com to inform them of the time this error occurred, and the actions you performed just before this error.
More information about this error may be available in the server error log.

**No errors inside PHP error log.**

**Inside the APACHE error log I see the following:** 

[Fri Jan 03 17:25:17.668262 2025] [:notice] [pid 19243:tid 4508972544] FastCGI: process manager initialized (pid 19243)
[Fri Jan 03 17:25:17.674074 2025] [mpm_event:notice] [pid 19242:tid 4508972544] AH00489: Apache/2.4.62 (Unix) OpenSSL/1.1.1w mod_fastcgi/mod_fastcgi-SNAP-0910052141 configured -- resuming normal operations
[Fri Jan 03 17:25:17.674801 2025] [core:notice] [pid 19242:tid 4508972544] AH00094: Command line: '/Applications/MAMP/Library/bin/httpd'
[Fri Jan 03 17:31:57.466795 2025] [fastcgi:error] [pid 19547:tid 123145344339968] [client ::1:60765] FastCGI: incomplete headers (0 bytes) received from server "/Applications/MAMP/fcgi-bin/php.fcgi"

**MySQL error log has the following:**

2025-01-03T22:25:17.6NZ mysqld_safe Logging to '/Applications/MAMP/logs/mysql_error.log'.
2025-01-03T22:25:17.6NZ mysqld_safe Starting mysqld daemon with databases from /Applications/MAMP/db/mysql80
2025-01-03T22:25:17.662184Z 0 [System] [MY-010116] [Server] /Applications/MAMP/Library/bin/mysql80/bin/mysqld (mysqld 8.0.40) starting as process 19241
2025-01-03T22:25:17.680718Z 0 [Warning] [MY-010159] [Server] Setting lower_case_table_names=2 because file system for /Applications/MAMP/db/mysql80/ is case insensitive
2025-01-03T22:25:17.735818Z 1 [System] [MY-013576] [InnoDB] InnoDB initialization has started.
2025-01-03T22:25:17.998016Z 1 [System] [MY-013577] [InnoDB] InnoDB initialization has ended.
2025-01-03T22:25:18.721702Z 0 [Warning] [MY-010068] [Server] CA certificate ca.pem is self signed.
2025-01-03T22:25:18.721791Z 0 [System] [MY-013602] [Server] Channel mysql_main configured to support TLS. Encrypted connections are now supported for this channel.
2025-01-03T22:25:18.769398Z 0 [System] [MY-011323] [Server] X Plugin ready for connections. Bind-address: '::' port: 33060, socket: /Applications/MAMP/tmp/mysql/mysqlx.sock
2025-01-03T22:25:18.769538Z 0 [System] [MY-010931] [Server] /Applications/MAMP/Library/bin/mysql80/bin/mysqld: ready for connections. Version: '8.0.40'  socket: '/Applications/MAMP/tmp/mysql/mysql.sock'  port: 8889  Source distribution.
2025-01-03T22:25:19.044904Z 8 [Warning] [MY-013360] [Server] Plugin mysql_native_password reported: ''mysql_native_password' is deprecated and will be removed in a future release. Please use caching_sha2_password instead'

I am using a fresh install of MAMP I only used to test out Expression Engine last week and moved all of its files out of the main folder.

**Installation:**
MAMP 7.2
MySQL 8.0.4
PHP 8.3.14

**Comment by givanz** [comment]
The Internal Server Error error message is generated by apache when php generates a fatal error, some possible causes can be missing php extensions like php-xml/dom, it's strange that no php error is logged.

For subfolder install you need to change `V_SUBDIR_INSTALL` in [/env.php#L46-L47](https://github.com/givanz/Vvveb/blob/master/env.php#L46-L47) to the name of the folder.

The error is displayed directly when accessing  http://localhost:8888/install/index.php or after pressing the install button?

My MAMP version is only 5.0.6 (7029) downloaded from https://www.mamp.info/en/windows/ maybe there are different versions and settings between Mac and Windows? 

![mamp](https://github.com/user-attachments/assets/e34beab1-a74e-422d-939f-ef105acd746a)

You can also try with XAMPP https://www.apachefriends.org/download.html or docker https://docs.docker.com/desktop/setup/install/mac-install/ the image is at https://hub.docker.com/repository/docker/vvveb/vvvebcms

**Reply by Coeur2louve** [reply]
Hello,
Same issue for me... do you succeed to fix the install ?
thank you

**Comment by Ezyweb-uk** [comment]
I'm trying to install in sub-folder /vvveb on XAMPP, I've set `defined('V_SUBDIR_INSTALL') || define('V_SUBDIR_INSTALL', '/vvveb');` in env.php, the install creates the database with all the tables (124), and then returns to /vvveb/install/index.php with error 500: End of script output before headers: index.php.

**Comment by givanz** [comment]
Please check your php error log and paste the last error messages.

One possible cause of interrupted install can be a low value for `max_execution_time` in php.ini you can increase the value and restart apache.

I just tested subdirectory install on xampp and it works, please make sure that you also update .htaccess and add the subdirectory path https://docs.vvveb.com/installation#subdirectory-install

![xampp](https://github.com/user-attachments/assets/b4a73f98-298a-48e5-a097-029318e87b1e)
